File: windows/RNTrackPlayer/Logic/Track.cs
using Microsoft.ReactNative.Managed;
using System;
using System.Diagnostics;

namespace TrackPlayer.Logic
{
    public class Track
    {
        public string Id { get; set; }
        public Uri Url { get; set; }
        public string Type { get; set; }
        public double Duration { get; set; }
        public string Title { get; set; }
        public string Artist { get; set; }
        public string Album { get; set; }
        public Uri Artwork { get; set; }

        private JSValueObject _originalObj;

        public Track(JSValueObject data)
        {
            Id = Utils.GetValue<string>(data, "id", null);
            Url = Utils.GetUri(data, "url", null);
            Type = Utils.GetValue<string>(data, "type", TrackType.Default);

            SetMetadata(data);

            _originalObj = data;
        }

        public void SetMetadata(JSValueObject data)
        {
            Title = Utils.GetValue<string>(data, "title", null);
            Artist = Utils.GetValue<string>(data, "artist", null);
            Album = Utils.GetValue<string>(data, "album", null);
            Artwork = Utils.GetUri(data, "artwork", null);
            Debug.WriteLine("Track.cs - implement merge of orig object");
        }

        public JSValueObject ToObject()
        {
            return _originalObj;
        }
    }

    public static class TrackType
    {
        public const string Default = "default";
        public const string Dash = "dash";
        public const string Hls = "hls";
        public const string SmoothStreaming = "smoothstreaming";
    }
}

File: windows/RNTrackPlayer/Logic/Track.cpp
#include "pch.h"
#include "Logic/Track.h"
#include "Logic/Utils.h"

using namespace winrt::RNTrackPlayer;

Track::Track(const React::JSValueObject& data)
    : Url(Utils::GetValue(data, "url", "")),
      Type(Utils::GetValue(data, "type", TrackType::Default))
{
     SetMetadata(data);
}

void Track::SetMetadata(const React::JSValueObject& data)
{
    Duration = std::chrono::seconds(data["duration"].AsInt64());
    Title = data["title"].AsString();
    Artist = data["artist"].AsString();
    Album = data["album"].AsString();
    Artwork = data["artwork"].AsString();
}

React::JSValueObject Track::ToObject()
{
    React::JSValueObject obj;

    obj["url"] = Url;
    obj["type"] = Type;

    obj["duration"] = Duration.count();

    obj["title"] = Title;
    obj["artist"] = Artist;
    obj["album"] = Album;
    obj["artwork"] = Artwork;

    return obj;
}
